File: Srims.WebSite/javascript/srims/documents/ContractType.js

if (!Srims.documents) 
    Ext.namespace('Srims.documents');

Ext.namespace('Srims.documents.ContractType');

Srims.documents.ContractType.MainContract = 'MainContract';
Srims.documents.ContractType.OutContract = 'OutContract';

Srims.documents.contractTypeRender = function(value){
    switch (value) {
        case 'MainContract':
            return '主合同';
        case 'OutContract':
            return '外协合同';
        default:
            return '未知';
    }
}
Srims.documents.contractTypeFilterItems = [{
    id: 'MainContract',
    text: '主合同'
}, {
    id: 'OutContract',
    text: '外协合同'
}];

Srims.documents.contractTypeStore = [['MainContract', '主合同'], ['OutContract', '外协合同']];
Srims.documents.contractTypeEditStore = [['MainContract', '主合同'], ['OutContract', '外协合同']];

File: Srims.Server.Business/Documents/ContractType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Srims.Server.Business.Documents
{
    /// <summary>
    /// 合同类型
    /// </summary>
    public enum ContractType
    {
        /// <summary>
        /// 未知类型
        /// </summary>
        Unkown = 0,
        /// <summary>
        /// 合同
        /// </summary>
        MainContract = 1,
        /// <summary>
        /// 外协合同
        /// </summary>
        OutContract = 2
    }
}
